### PR TITLE
Add Format Number implementation and tests

### PR DESCRIPTION
**Pull request by marcinkrykowski**

### CODE REVIEW

File: java-numbers/src/main/java/com/baeldung/formatNumber/FormatNumber.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.Locale;
+
+public class FormatNumber {
+        private static final double D = 4.2352989244d;
+        private static final double F = 8.6994540927d;
+
+        public static void main(String[] args) {
```
Comment (glmartin) [inline java-numbers/src/main/java/com/baeldung/formatNumber/FormatNumber.java:13]: Let's remove the main method, and only keep the methods that do the formatting.

File: java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.Test;
+
+import static com.baeldung.formatNumber.FormatNumber.*;
+import static org.assertj.core.api.Assertions.*;
+
+public class FormatNumberUnitTest {
+        private static final double D = 4.2352989244d;
+        private static final double F = 8.6994540927d;
+
+        @Test public void givenDecimalNumber_whenFormatNumberToNDecimalPlaces_thenGetExpectedResult() {
```
Comment (glmartin) [inline java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java:12]: Let's move the method below the `@Test` annotation

File: java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package com.baeldung.formatNumber;
+
+import org.junit.Test;
+
+import static com.baeldung.formatNumber.FormatNumber.*;
+import static org.assertj.core.api.Assertions.*;
+
+public class FormatNumberUnitTest {
+        private static final double D = 4.2352989244d;
```
Comment (glmartin) [inline java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java:9]: Indentation should be only 4 spaces

File: java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                assertThat(withBigDecimal(D, 2)).isEqualTo(4.24);
+                assertThat(withBigDecimal(D, 3)).isEqualTo(4.235);
+                assertThat(withBigDecimal(F, 2)).isEqualTo(8.7);
+                assertThat(withBigDecimal(F, 3)).isEqualTo(8.699);
+
+                assertThat(withMathRound(D, 2)).isEqualTo(4.24);
+                assertThat(withMathRound(D, 3)).isEqualTo(4.235);
+                assertThat(withMathRound(F, 2)).isEqualTo(8.7);
+                assertThat(withMathRound(F, 3)).isEqualTo(8.699);
+
```
Comment (glmartin) [inline java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java:23]: We should split up the tests into separate `@Test` methods for each solution.  For example, there should be a `@Test` method for `withBigDecimal`, `withMathRound`, `withStringFormat` , `withDecimalFormatLocal` , and`withDecimalFormatPattern` .

File: java-numbers/src/main/java/com/baeldung/formatNumber/FormatNumber.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
+package com.baeldung.formatNumber;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.Locale;
+
+public class FormatNumber {
+        public static double withBigDecimal(double value, int places) {
```
Comment (glmartin) [inline java-numbers/src/main/java/com/baeldung/formatNumber/FormatNumber.java:10]: Indentation should be 4 spaces.

File: java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java (hunk 2 above)
Comment (marcinkrykowski) [reply java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java:12]: Sure

File: java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java (hunk 3 above)
Comment (marcinkrykowski) [reply java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java:9]: Opss.. Done

File: java-numbers/src/main/java/com/baeldung/formatNumber/FormatNumber.java (hunk 5 above)
Comment (marcinkrykowski) [reply java-numbers/src/main/java/com/baeldung/formatNumber/FormatNumber.java:10]: Done

File: java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java (hunk 4 above)
Comment (marcinkrykowski) [reply java-numbers/src/test/java/com/baeldung/formatNumber/FormatNumberUnitTest.java]: Okay

**Comment by marcinkrykowski** [comment]
@glmartin I finished what you suggested. Thanks for this! I also move it to java-numbers-3 module 👍

**Pull request merged**